lib/Barriers/PreDefined.pm

```diff
@@ -113,49 +125,93 @@ The number of the display decimal
 
 The method for the barrier calculation, method_1 or method_2
 
-=head2 config_param
+=head2 contract_barrier_levels
 
-A configuration parameter contains the configuration of each contract type
+A set of barrier level that intended to obtain for a contract type
 
 =cut
 
-my $config_param = LoadFile(File::ShareDir::dist_file('Barriers::PreDefined', 'contract_type_config.yml'));
+has [qw(
+       config_file
+       contract_type
+       duration
+       central_spot
+       display_decimal
+       method
+       )
+     ] => (
+     is  => 'ro'
+    );
+
+has contract_barrier_levels => (
+      is => 'rw',
+      lazy_build => 1,
+);
+
+
+sub _build_contract_barrier_levels {
+    my $self = shift;
+
+    my $config = $self->config_file;
+
+    my $contract_barrier_levels;
+
+    for my $set (@$config) {
+    for my $type (@{ $set->{types} }) {
+        $contract_barrier_levels->{$type} = $set->{levels}
+    }
+}
+
+   return $contract_barrier_levels;
+}
+
 
 
 =head1 METHODS
 
-calculate_available_barriers($contract_type, $duration, $central_spot, $display_decimal, $method);
+=cut
+=head2 calculate_available_barriers
 
 A function to calculate available barriers for a contract type
-
+Input_parameters: $contract_type, $duration, $central_spot, $display_decimal, $method
 
 =cut
 
 sub calculate_available_barriers {
-    my ($contract_type, $duration, $central_spot, $display_decimal, $method) = @_;
+    my $args = shift;
+
+    my ($contract_type, $duration, $central_spot, $display_decimal, $method) = @{$args}{qw(contract_type duration central_spot display_decimal method)};
```

Comment:
as we discussed, we don't need `method` (will be separate configs)

Also, please use the previous version of unpacking arguments, i.e. 

```perl
my ($contract_type, $duration, $central_spot, $display_decimal) = @_;
```